app/src/main/java/org/wikipedia/views/FaceAndColorDetectImageView.kt

```diff
@@ -0,0 +1,84 @@
+package org.wikipedia.views
+
+import android.content.Context
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.Drawable
+import android.net.Uri
+import android.util.AttributeSet
+import androidx.annotation.DrawableRes
+import androidx.appcompat.widget.AppCompatImageView
+import androidx.palette.graphics.Palette
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.DataSource
+import com.bumptech.glide.load.MultiTransformation
+import com.bumptech.glide.load.engine.GlideException
+import com.bumptech.glide.load.resource.bitmap.CenterCrop
+import com.bumptech.glide.load.resource.bitmap.DownsampleStrategy
+import com.bumptech.glide.request.RequestListener
+import com.bumptech.glide.request.target.Target
+import org.apache.commons.lang3.StringUtils
+import org.wikipedia.settings.Prefs
+import org.wikipedia.util.CenterCropWithFaceTransformation
+import org.wikipedia.util.WhiteBackgroundTransformation
+import java.util.*
+
+class FaceAndColorDetectImageView constructor(context: Context, attrs: AttributeSet? = null) : AppCompatImageView(context, attrs) {
+
+    interface OnImageLoadListener {
+        fun onImageLoaded(palette: Palette)
+        fun onImageFailed()
+    }
+
+    private fun shouldDetectFace(uri: Uri): Boolean {
+        // TODO: not perfect; should ideally detect based on MIME type.
+        val path = StringUtils.defaultString(uri.path).toLowerCase(Locale.ROOT)
```

Comment:
Usages of `StringUtils.defaultString()` can be replaced with `.orEmpty()`

Reply:
Correct! Thanks.